code/modules/atmospherics/machinery/components/fusion/hfr_parts.dm

```diff
@@ -357,14 +357,15 @@
 			connected_core.waste_remove = !connected_core.waste_remove
 			. = TRUE
 		if("filter")
-			connected_core.filter_type = null
-			var/filter_name = "nothing"
-			var/gas = gas_id2path(params["mode"])
-			if(gas in GLOB.meta_gas_info)
-				connected_core.filter_type = gas
-				filter_name = GLOB.meta_gas_info[gas][META_GAS_NAME]
-			investigate_log("was set to filter [filter_name] by [key_name(usr)]", INVESTIGATE_ATMOS)
+			connected_core.moderator_scrubbing ^= gas_id2path(params["mode"])
 			. = TRUE
+		if("mod_filtering_rate")
+			var/mod_filtering_rate = params["mod_filtering_rate"]
+			if(text2num(mod_filtering_rate) != null)
+				mod_filtering_rate = text2num(mod_filtering_rate)
+				. = TRUE
+			if(.)
```

Comment:
what's the point of this if check?

Reply:
i remember old thermomachines uses that for ensure that the number passed was right and not somehow bugged/cheated or something, so i used that on this, tho i don't know if is really needed

Reply:
You should clamp the user input between the acceptable filtering rates, but if . is only set to true in an if block one line up, you can just cut out the middleman. . = TRUE is useful, it makes the ui update, but checking it isn't needed

Reply:
aight

Reply:
i did the same to the other parts in the same proc, for consistency